app/src/test/java/org/mozilla/tv/firefox/pinnedtile/PinnedTileTest.kt

```diff
@@ -0,0 +1,104 @@
+import androidx.test.core.app.ApplicationProvider
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.reactivex.Observable
+import io.reactivex.android.plugins.RxAndroidPlugins
+import io.reactivex.observers.TestObserver
+import io.reactivex.plugins.RxJavaPlugins
+import io.reactivex.schedulers.Schedulers
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.tv.firefox.focus.FocusRepo
+import org.mozilla.tv.firefox.navigationoverlay.NavigationOverlayViewModel
+import org.mozilla.tv.firefox.navigationoverlay.channels.ChannelDetails
+import org.mozilla.tv.firefox.session.SessionRepo
+import org.mozilla.tv.firefox.utils.FormattedDomainWrapper
+import org.robolectric.RobolectricTestRunner
+
+/**
+ * Unit tests for pinned tile operations in [NavigationOverlayViewModel].
+ */
+@RunWith(RobolectricTestRunner::class)
+class PinnedTileTest {
+
+    companion object {
+        @BeforeClass
+        @JvmStatic
+        fun beforeClass() {
+            RxJavaPlugins.setComputationSchedulerHandler { Schedulers.trampoline() }
+            RxAndroidPlugins.setInitMainThreadSchedulerHandler { Schedulers.trampoline() }
+        }
+    }
+
+    @MockK private lateinit var sessionRepo: SessionRepo
+    @MockK private lateinit var focusRepo: FocusRepo
+    @MockK private lateinit var pinnedTileImageUtilWrapper: PinnedTileImageUtilWrapper
+    @MockK private lateinit var formattedDomainWrapper: FormattedDomainWrapper
+    @MockK private lateinit var drawable: Drawable
+    private lateinit var overlayVm: NavigationOverlayViewModel
+    private lateinit var pinnedTileRepo: PinnedTileRepo
+    private lateinit var testObserver: TestObserver<ChannelDetails>
+
+    @Suppress("DEPRECATION")
+    @Before
+    fun setUp() {
+        MockKAnnotations.init(this)
+        every { focusRepo.focusUpdate } answers { Observable.empty() }
+        every { focusRepo.defaultViewAfterScreenChange } answers { Observable.empty() }
+        every { sessionRepo.legacyState } answers { MutableLiveData() }
+        every { pinnedTileImageUtilWrapper.generatePinnedTilePlaceholder(any()) } answers { drawable }
+        every { formattedDomainWrapper.format(any(), any(), any()) } answers { "" }
+
+        pinnedTileRepo = PinnedTileRepo(ApplicationProvider.getApplicationContext())
+        overlayVm = NavigationOverlayViewModel(
+                sessionRepo,
+                focusRepo,
+                pinnedTileImageUtilWrapper,
+                formattedDomainWrapper,
+                pinnedTileRepo
+        )
+        testObserver = overlayVm.pinnedTiles.test()
+    }
+
+    @Test
+    fun `WHEN repo emits a successful load THEN view model should emit a list of same number of tiles`() {
+        assertEquals(1, testObserver.valueCount())
+        assertEquals(10, testObserver.values().last().tileList.size)
```

Comment:
Oh for a second I was confused, but this 10 is the number of default pinned tiles that we ship already. Can you add a comment here?

Reply:
Replaced with a constant.